core/src/main/java/com/scalar/db/util/groupcommit/Slot.java

```diff
@@ -0,0 +1,123 @@
+package com.scalar.db.util.groupcommit;
+
+import com.google.common.base.MoreObjects;
+import com.scalar.db.util.ThrowableRunnable;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+// A container of value which is stored in a group.
+@ThreadSafe
+class Slot<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V> {
+  private final AtomicReference<Group<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V>> parentGroup =
+      new AtomicReference<>();
+  private final CHILD_KEY key;
+  // If a result value is null, the value is already emitted.
+  // Otherwise, the result lambda must be emitted by the receiver's thread.
+  private final CompletableFuture<ThrowableRunnable> completableFuture = new CompletableFuture<>();
```

Comment:
```suggestion
  private final CompletableFuture<ThrowableRunnable<Exception>> completableFuture = new CompletableFuture<>();
```

Reply:
Thanks! Fixed it in https://github.com/scalar-labs/scalardb/pull/1607/commits/865830013b893112149fa88ce705cd60aa465226